File: Chronir-iOS/Sources/DesignSystem/Tokens/AnimationTokens.swift
import SwiftUI

// Values sourced from design-tokens/tokens/animation.json (Style Dictionary)
// See docs/design-system.md Section 3.6 for full spec

enum AnimationTokens {
    // MARK: - Durations

    static let instant: Animation = .easeInOut(duration: 0.1)
    static let fast: Animation = .easeInOut(duration: 0.15)
    static let standard: Animation = .easeInOut(duration: 0.3)
    static let slow: Animation = .easeInOut(duration: 0.5)

    // MARK: - Springs

    static let spring: Animation = .spring(
        response: 0.35,
        dampingFraction: 0.8
    )
    static let gentleSpring = spring  // alias for backward compat

    static let bouncySpring: Animation = .spring(
        response: 0.4,
        dampingFraction: 0.65
    )

    // MARK: - Raw Durations (for non-Animation contexts)

    static let instantDuration: Double = 0.1
    static let fastDuration: Double = 0.15
    static let standardDuration: Double = 0.3
    static let slowDuration: Double = 0.5
}

File: Chronir-Android/core/designsystem/src/main/java/com/chronir/designsystem/tokens/AnimationTokens.kt
package com.chronir.designsystem.tokens

// Values sourced from design-tokens/tokens/animation.json (Style Dictionary)
// See docs/design-system.md Section 3.6 for full spec

object AnimationTokens {
    // MARK: - Durations (milliseconds)

    const val DurationInstant = 100
    const val DurationFast = 150
    const val DurationStandard = 300
    const val DurationSlow = 500

    // MARK: - Spring

    const val SpringDampingRatio = 0.8f
    const val SpringStiffness = 400f

    const val BouncySpringDampingRatio = 0.65f
    const val BouncySpringStiffness = 300f

    // Backward compat aliases
    const val DurationMedium = DurationStandard
    const val DurationVerySlow = 700
}
